lib/cn/move.c

```diff
@@ -101,6 +102,7 @@ cn_move(
             kvset_set_compc(src->le_kvset, compc);
 
             src = src_next;
+            src_cnt--;
```

Comment:
The bug was that we obtained src_end from a naked traversal of the root list (above), but src_end might not exist or  no longer be on the list by the time we get the tree write lock here, and so this loop would go off into the weeds...